index.js

```diff
@@ -3044,81 +3044,151 @@
     impl: at
   };
 
-  //# head :: Array a -> Maybe a
+  //# head :: Foldable f => f a -> Maybe a
   //.
-  //. Returns Just the first element of the given array if the array contains
-  //. at least one element; Nothing otherwise.
+  //. Returns Just the first element of the given collection if the foldable
+  //. contains at least one element; Nothing otherwise.
```

Comment:
We avoid the term _collection_ because people tend not to think of values of types such as `Identity a`, `Maybe a`, and `Either a b` as collections.

I think we should also avoid _foldable_. Referring to a type class works in this case, but this approach would be problematic for functions which place multiple constraints on a single type variable. Let's avoid referring to _applicative semigroups_ and _foldable applicative monoids_. :joy:

_Structure_ is the best term we have considered thus far.

Reply:
🆗

Reply:
I understand this now.

Reply:
> Returns Just the first element of the given structure if *it* contains at least one element; Nothing otherwise.

This is ambiguous: _it_ could refer to _the given structure_ or to _the first element of the given structure_. Let's use _the structure_ in place of _it_ to be clear, even though it's repetitive.

Reply:
🆗